WebServer/server_first.py

```diff
@@ -0,0 +1,49 @@
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from http.cookies import SimpleCookie
+
+from read_page_inf import read_page
+from error_handler import error_page
+
+
+class ServerFirst(BaseHTTPRequestHandler):
+
+    def do_GET(self):
+        if self.path in self.function_of_page:
+            self.function_of_page[self.path](self)
+        else:
+            error_page(self, self.path)
+
+    def hello_page(self):
+        response, page = read_page('http://localhost:8001/hello_page.html')
```

Comment:
May be I'm missing something but why are you using this
```python
response, page = read_page('http://localhost:8001/hello_page.html')
```
and not this
```python
response, page = read_page('hello_page.html')
```
?

So you could get rid of `s_path[21:]` in ` page_to_open = open('pages' + s_path[21:], 'r').read()`

Reply:
You are right. It didn't make any special sense. Corrected